src/shell/mod.rs

```diff
@@ -416,13 +416,23 @@ impl WorkspaceSet {
             return Err(InvalidWorkspaceIndex);
         }
 
+        // Animate if workspaces overview isn't open
+        let layer_map = layer_map_for_output(&self.output);
+        let animate = !layer_map
+            .layers()
+            .any(|l| l.namespace() == WORKSPACE_OVERVIEW_NAMESPACE);
+
         if self.active != idx {
             let old_active = self.active;
             state.remove_workspace_state(&self.workspaces[old_active].handle, WState::Active);
             state.remove_workspace_state(&self.workspaces[old_active].handle, WState::Urgent);
             state.remove_workspace_state(&self.workspaces[idx].handle, WState::Urgent);
             state.add_workspace_state(&self.workspaces[idx].handle, WState::Active);
-            self.previously_active = Some((old_active, workspace_delta));
+            self.previously_active = if animate {
+                Some((old_active, workspace_delta))
+            } else {
+                None
+            };
```

Comment:
Same question as previously, how does this interact with gestures?

Looks like it should just immediately swap the workspace, which seems a bit jarring. We probably want the windows from the overview to react to the gesture, so do we need a way further down the line to capture the gesture in the overview-client? I guess for now it might make sense to simply disable the gesture, when the overview is visible?

Reply:
Conclusion from mattermost: We should disabled gestures while the workspace overview is open.